toolchains/cpu.spack.gyselalibxx_env/gyselalibxx-env-0.23.1.yaml

```diff
@@ -8,7 +8,6 @@ spack:
     - 'fftw@3.3 precision=float,double'
     - 'ginkgo@1.8'
     - 'googletest@1.14 +gmock'
-    - 'kokkos@4.4.1:4.4'
```

Comment:
I don't understand this change. You said in #343 that you had 2 versions of kokkos in the environment but here there is only one. Where did the second come from?

If we can keep this I would prefer it as the compilation of kokkos is a non-negligible part of our build time.

Reply:
Is there a more recent spack that provides Kokkos 4.5.1 and higher? The only reasons I know to prefer one spack version over another are:
- Higher versions have more recent packages
- Specific versions are sometimes required to match the version on a specific cluster

This toolchain is not designed for clusters so I think we can neglect the second argument.

Reply:
Not yet, the next version, Spack v1, have been delayed to the end of June.

Reply:
It is not required, there is a configuration where it works: Kokkos 4.4.1 with Kokkos Kernels 4.5.1. We can lower the minimal requirement to 4.4.1, it was in fact my proposition 1.